File: BankingApplicationSolution/BankingLibrary/InterestAccount.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BankingLibrary {
    
    public class InterestAccount : Account {

        public decimal InterestRate { get; private set; }

        public void CalculateInterest(int Months) {
            var interest = this.Balance * (this.InterestRate / 12) * Months;
            Deposit(interest);
            Console.WriteLine($"Calculated interest is {interest}");
        }

        public override string ToString() {
            return $"{base.ToString()}, IntRate={InterestRate}";
        }
        public InterestAccount(double interestRate) : this() {
            InterestRate = Convert.ToDecimal(interestRate);
        }
        public InterestAccount() : base() {
            this.Description = "Interest Account";
        }
    }
}

File: BankingInJava/src/com/maxtrain/bootcamp/dotnet/dnbc8/InterestAccount.java
package com.maxtrain.bootcamp.dotnet.dnbc8;

class InterestAccount extends Account {

	private double intRate;
	
	public void print() {
		System.out.println(this.toString());
	}

	public InterestAccount(double intRate, String description) {
		super(description);
		this.setIntRate(intRate);
	}
	
	public InterestAccount(double intRate) {
		this(intRate, "New InterestAccount");
	}
	
	public InterestAccount() {
		this(1.0);
	}

	public double getIntRate() {
		return intRate;
	}
	public void setIntRate(double intRate) {
		this.intRate = intRate;
	}

}
